app/models/account.rb

```diff
@@ -90,7 +90,7 @@ class Account < Ekylibre::Record::Base
   validates :last_letter, length: { allow_nil: true, maximum: 10 }
   validates :name, length: { allow_nil: true, maximum: 200 }
   validates :number, uniqueness: true
-  validates :number, length: { is: 8 }, format: { without: /\A0/ }, unless: :already_existing_and_general
+  validates :number, length: { is: Preference[:account_number_digits] }, format: { without: /\A0/ }, unless: :already_existing_and_general
```

Comment:
This doesn't work. You need to put this inside a block or it won't get evaluated at validation time.